### PR TITLE
Change　grules in sky130_mapped file

### PR DESCRIPTION
**Pull request by circuit4**
Removed
grulesobj["mcon"]["poly"] = {"min_enclosure": 0.08}
grulesobj["mcon"]["met1"] = {"min_enclosure": 0.08}
grulesobj["via1"]["met2"] = {"min_enclosure": 0.06}
grulesobj["via2"]["met3"] = {"min_enclosure": 0.09}
grulesobj["via3"]["met4"] = {"min_enclosure": 0.65}
grulesobj["via4"]["met5"] = {"min_enclosure": 0.07}

 Added
grulesobj["poly"]["mcon"] = {"min_enclosure": 0.08, "min_separation": 0.06} #licon.8a
grulesobj["met1"]["mcon"] = {"min_enclosure": 0.08} #li.5.-
grulesobj["met2"]["via1"] = {"min_enclosure": 0.06} #m1.5
grulesobj["met3"]["via2"] = {"min_enclosure": 0.09} #m2.5
grulesobj["met4"]["via3"] = {"min_enclosure": 0.07} #m3.4 (as previously stated)
grulesobj["met5"]["via4"] = {"min_enclosure": 0.07} #m4.3

For example,  via3 will never enclose met4 so even though the value 0.65 is 10x bigger than it should be it will never be checked.
The other areas that were modified are for similar reasons.

Therefore , I modified grules .

### CODE REVIEW

**Comment by alibillalhammoud** [comment]
Hello @circuit4 


The grules are actually bidirectional, so it doesn't matter what order you specify them.  
Basically, we simplify the rules by saying that the rules can be represented by an undirected graph. This is good enough for almost all cases. 

The pdk class will return which ever rule it finds regardless of the order you pass the layers into `pdk.get_grule()`

File: openfasoc/generators/glayout/glayout/flow/pdk/sky130_mapped/grules.py
Comment (alibillalhammoud) [inline openfasoc/generators/glayout/glayout/flow/pdk/sky130_mapped/grules.py:1]: Looking at the GitHub commit action this seems to generate DRC errors for the opamp. 

@chetanyagoyal  you want to take a look?

**Review by alibillalhammoud — CHANGES_REQUESTED** [review]

File: openfasoc/generators/glayout/glayout/flow/pdk/sky130_mapped/grules.py
Comment (chetanyagoyal) [reply openfasoc/generators/glayout/glayout/flow/pdk/sky130_mapped/grules.py:1]: yes I will take a closer look

Comment (msaligane) [reply openfasoc/generators/glayout/glayout/flow/pdk/sky130_mapped/grules.py:1]: Thanks Ali. @chetanyagoyal any updates?

**Comment by alibillalhammoud** [comment]
Thank you again for the PR. @circuit4 @chetanyagoyal Were you able to find anything? I think one point @circuit4 brought up which may be worth exploring is that some rules are larger than they should be.

@circuit4 if this has been resolved feel free to close this PR, otherwise we can keep open or maybe make this a draft PR

**Pull request closed without merging**